pandas/tests/reshape/merge/test_merge.py

```diff
@@ -2409,3 +2409,23 @@ def test_merge_result_empty_index_and_on():
 
     result = merge(df2, df1, left_index=True, right_on=["b"])
     tm.assert_frame_equal(result, expected)
+
+
+def test_merge_suffixes_produce_dup_columns_warns():
+    # GH#22818
+    left = DataFrame({"a": [1, 2, 3], "b": 1, "b_x": 2})
+    right = DataFrame({"a": [1, 2, 3], "b": 2})
+    with tm.assert_produces_warning(FutureWarning):
+        merge(left, right, on="a")
+
+    with tm.assert_produces_warning(FutureWarning):
+        merge(right, left, on="a", suffixes=("_y", "_x"))
+
+
+def test_merge_duplicate_columns_with_suffix_no_warning():
+    # GH#22818
+    left = DataFrame([[1, 1, 1], [2, 2, 2]], columns=["a", "b", "b"])
+    right = DataFrame({"a": [1, 3], "b": 2})
+    result = merge(left, right, on="a")
+    expected = DataFrame([[1, 1, 1, 2]], columns=["a", "b_x", "b_x", "b_y"])
```

Comment:
is this the expected result?

Reply:
Yes, I wanted to avoid raising when we already get duplicate columns as input, because the collisions in this case are not caused by the suffixes

Reply:
kk, can you add a comment to this effect

Reply:
Will do.